### ISSUE TITLE
[Parser] Issues with `BoolOp` node generation in new parser

### ISSUE THREAD

**Issue by akshanshbhatt** [lcompilers/lpython#777]
```py
❯ cat examples/expr2.py
a or b or c or d

❯ lpy --show-ast examples/expr2.py --tree  # Existing Parser
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-Name
  |       | |-id=a
  |       | └-expr_contextType=Load
  |       |-Name
  |       | |-id=b
  |       | └-expr_contextType=Load
  |       |-Name
  |       | |-id=c
  |       | └-expr_contextType=Load
  |       └-Name
  |         |-id=d
  |         └-expr_contextType=Load
  └-type_ignores=↧

❯ lpy --show-ast --new-parser examples/expr2.py --tree  # New Parser
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-BoolOp
  |       | |-boolopType=Or
  |       | └-values=↧
  |       |   |-BoolOp
  |       |   | |-boolopType=Or
  |       |   | └-values=↧
  |       |   |   |-Name
  |       |   |   | |-id=a
  |       |   |   | └-expr_contextType=Load
  |       |   |   └-Name
  |       |   |     |-id=b
  |       |   |     └-expr_contextType=Load
  |       |   └-Name
  |       |     |-id=c
  |       |     └-expr_contextType=Load
  |       └-Name
  |         |-id=d
  |         └-expr_contextType=Load
  └-type_ignores=↧
```

The new parser currently only supports `BoolOp` node generation with two operands (like `BinOp`). If an expression contains multiple `and` or `or`, the parser keeps creating new `BoolOp` nodes instead of retaining them in a single node with multiple `expr` values.

**Comment by Thirumalai-Shaktivel** [comment]
Some issues are fixed in #773. The only left out part is:
```py
(a or b) or c
```
The problem here is that, as parenthesis information is not used or stored by the macros, it is difficult to handle the above example:
OLD Parser AST:
```console
$ lpython --show-ast examples/expr2.py --tree
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-BoolOp
  |       | |-boolopType=Or
  |       | └-values=↧
  |       |   |-Name
  |       |   | |-id=a
  |       |   | └-expr_contextType=Load
  |       |   └-Name
  |       |     |-id=b
  |       |     └-expr_contextType=Load
  |       └-Name
  |         |-id=c
  |         └-expr_contextType=Load
  └-type_ignores=↧
```
NEW Parser AST (this PR):
```console
$ lpython --show-ast examples/expr2.py --tree --new-parser
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-Name
  |       | |-id=a
  |       | └-expr_contextType=Load
  |       |-Name
  |       | |-id=b
  |       | └-expr_contextType=Load
  |       └-Name
  |         |-id=c
  |         └-expr_contextType=Load
  └-type_ignores=↧
```

**Comment by Thirumalai-Shaktivel** [comment]
Closing in favour of #848.
